### ISSUE TITLE
stagger folder/file name columns in folder grouping view

### ISSUE THREAD

**Issue by Efreak** [nicotine-plus/nicotine-plus#1861]
### Describe the solution you'd like
When viewing Downloads in folder-grouped view, please don't waste the space used by the path column. Overlap the filename and directory name columns, whenever they aren't both being used in a single view. The filenames should be offset to make them obvious, and probably still have their own header for sorting.

### Additional context
Basically, the path column should overflow into the filename column (and beyond) *except* when said columns past the path column have text in them. The path column should be on top of the other columns backgrounds, but below their contents, if that makes sense.

Current View:
[![current preview](https://i.imgur.com/rR9Jl9wm.png)](https://i.imgur.com/rR9Jl9w.png)
Proposed View: 
[![proposed preview](https://i.imgur.com/wHlHDxom.png)](https://i.imgur.com/wHlHDxo.png)

**Comment by mathiascode** [comment]
I don't think this is possible in GTK (but let me know if there's a way). You would probably need a combined column similar to https://github.com/nicotine-plus/nicotine-plus/issues/1494.

**Comment by Efreak** [comment]
You'd have to have a 'display' value and an internal value for each field, so that you can display `\t$filename` for files when the internal value is `$folder/$filename`, and still be able to sort properly. In this case, you'd have *only* a combined column visible for this view, which wouldn't display in other views. I don't know if this is possible in GTK or not, I do know it's possible in other languages. For example, an html/js funtion I've used to sort by hidden criteria is `HTMLUListElement.prototype.sort=function(){[...this.children].sort((a,b)=>parseInt(a.dataset.value)>parseInt(b.dataset.value)).forEach(li=>this.appendChild(li))}`.

**Comment by mathiascode** [comment]
A "hidden" column with values for sorting is not an issue if we add a combined/compact column view in the future (except for maybe a minor performance hit and somewhat increased memory usage), we already use those, but a short-term solution of overflowing columns in the current view is not possible by default in GTK, afaik.

**Comment by mathiascode** [comment]
This request is blocked until we've ported the search/transfer lists to the new TreeView class.
